index.js

```diff
@@ -1,40 +1,74 @@
 // ***************************************************************************
 // Iteration 1 - `for...of` loop
 // ***************************************************************************
-
+console.log("===== Iteration 1 ====== ");
 const getFirstNames = arr => {
   const userFirstNames = [];
   for (let user of arr) {
     // Your code goes here ...
+    userFirstNames.push(user.firstName);
   }
+  return userFirstNames;
 };
 
-getFirstNames(usersArray);
+console.log(getFirstNames(usersArray));
 // expected output:
 // [ 'Kirby', 'Tracie', 'Kendra', 'Kinney', 'Howard', 'Rachelle', 'Lizzie' ]
 
 // ***************************************************************************
 // Iteration 2 - `for...of` loop and ES6 string literals `${}`
 // ***************************************************************************
-
+console.log("===== Iteration 2 ====== ");
 const getFullNames = arr => {
+  const userFullNames = [];
   // Your code goes here ...
+  for (let user of arr) {
+    userFullNames.push(user.firstName + " " + user.lastName);
+  }
+  return userFullNames;
 };
 
-getFullNames(usersArray);
+console.log(getFullNames(usersArray));
 // expected output:
 // [ 'Kirby Doyle', 'Tracie May', 'Kendra Hines', 'Kinney Howard',
 //   'Howard Gilmore', 'Rachelle Schneider', 'Lizzie Alford' ]
 
 // ***************************************************************************
 // Iteration 3 - ES6 destructuring , for of loop, object literal
 // ***************************************************************************
-
+console.log("===== Iteration 3 ====== ");
+// const getUsersCreditDetails = arr => {
+//   // Your code goes here ...
+//   let usersCreditDetails = [];
+//   for (let user of arr) {
+//     let obj = {};
+//     obj.firstname = user.firstName;
+//     obj.lastName = user.lastName;
+//     obj.balance = user.balance;
+//     usersCreditDetails.push(obj);
+//   }
+//   return usersCreditDetails;
+// };
 const getUsersCreditDetails = arr => {
   // Your code goes here ...
-};
+  let usersCreditDetails = [];
+  for ({
+      firstName,
+      lastName,
+      balance
+    } of arr) {
+    // let  {firstname,lastName,balance} = user;
+
+    usersCreditDetails.push({
+      firstName,
+      lastName,
+      balance
+    });
```

Comment:
this is quite clean! good job.

The formatting throws it away a little.
For better formatting you could have user a variable - user, for example - in the for loop and immediately after destrcture them